addons/base_vat/tests/test_validate_ruc.py

```diff
@@ -14,3 +14,24 @@ def test_peru_ruc_format(self):
         with self.assertRaises(ValidationError):
             partner.vat = '11111111111'
         partner.vat = '20507822470'
+
+    def test_parent_validation(self):
+        """Test the validation with company and contact"""
+
+        # disable the verification to set an invalid vat number
+        self.env.user.company_id.vat_check_vies = False
+        company = self.env["res.partner"].create({
+            "name": "World Company",
+            "country_id": self.env.ref("base.be").id,
+            "vat": "BE0123456789",
+            "company_type": "company",
+        })
+        contact = self.env["res.partner"].create({
+            "name": "Sylvestre",
+            "parent_id": company.id,
+            "company_type": "person",
+        })
+
+        # reactivate it and correct the vat number
+        self.env.user.company_id.vat_check_vies = True
```

Comment:
Isn't this going to red runbot because the service can't be reached?

Reply:
how so? looks green

Reply:
If `vat_check_vies` is `True`, it means that the runbot will do a VIES check, right? Therefore, call an external service. This is usually likely to cause timeouts if the service is unreachable. By the way, isn't there a risk of the runbot IP being blacklisted if they do too many requests?

Reply:
Oh right, let's mock that

Reply:
indeed, better